Zhou Yang ip

SamuelZero1 opened the pull request:
SamuelBot Pro
“Your mind is for having ideas, not holding them.” – David Allen

SamuelBot Pro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
FAST SUPER FAST to use
All you need to do is,
download it from [here](https://nus-cs2103-ay2324s2.github.io/website/admin/ip-w4.html).
double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is FREE!

Features:

 Managing tasks
 Managing deadlines 
 Managing events

 Reminders (coming soon)
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

public class Main {
    public static void main(String[] args) {
        Application.launch(ChatBotGUI.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Duke Level0

Commit:
Duke_Level1

Commit:
Duke_Level2

Commit:
Duke_Level3

Commit:
Duke_Level4

Commit:
Duke_Level5

ReflectiveObsidian commented on src/main/java/Duke_Level4.java:
```diff
@@ -0,0 +1,85 @@
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("What can I do for you?");
+        List<String> list = new ArrayList<>();
```
Could this variable name be updated to show that there are multiple objects represented? 
```suggestion
        List<String> tasks = new ArrayList<>();
```

ReflectiveObsidian reviewed (commented):
Just one nitpick.

KohGuanZeh commented on src/main/java/Duke_Level4.java:
```diff
@@ -0,0 +1,85 @@
+        for (int i = 0; i < task.length; i++) {
+            task[i] = "[]";
+        }
+        String[] done = new String[100];
```
Maybe this variable can use a better name such as `taskPrefixes`?

KohGuanZeh commented on src/main/java/Duke_Level4.java:
```diff
@@ -0,0 +1,85 @@
+        }
+        Scanner s = new Scanner(System.in);
+        String str = s.nextLine();
+        while(! str.equals("bye")){
+            if(str.substring(0, 4).equals("mark")){
```
I think there should be a space in between `if`, `while` and the brackets?

KohGuanZeh commented on src/main/java/Duke_Level1.java:
```diff
@@ -0,0 +1,14 @@
+        System.out.println("What can I do for you?");
+        Scanner s = new Scanner(System.in);
+        String str = s.nextLine();
+        while(! str.equals("bye")) {
```
I think there should not be a space between `!` and `str.equals("bye")`?

KohGuanZeh commented on src/main/java/Duke_Level4.java:
```diff
@@ -0,0 +1,85 @@
+import java.util.ArrayList;
```
Perhaps consider having a package for `Duke`?

KohGuanZeh reviewed (commented):
I see that you have been making progress with the program as you are starting to incorporate more classes to better manage your programs. I think that the program can be further improved by using names more representative of what the variable is.

Commit:
Duke_Level5

Commit:
Duke_Level6

Commit:
Duke

Commit:
Duke_Level7

Commit:
Duke_Level8

Commit:
ui

Commit:
Duke_Level8

Commit:
Duke_Level8

Commit:
A_Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Junit

Commit:
Update

Commit:
Duke_Level9

Commit:
Duke_Level9

Commit:
SamuelBot

Commit:
SamuelBot

Commit:
Merge branch 'A-CodingStandard'

# Conflicts:
#	SamuelBot.txt
#	src/main/java/SamuelBot/Ui.java

Commit:
Duke_Level10

Commit:
Duke_Level10

Commit:
Duke_Level10

Commit:
GUI

Commit:
Duke_Level10

Commit:
Duke_Level9
Commit: Add feature to search for items in inventory

Description:
This commit adds the functionality to search for items in the inventory. Users can now enter keywords in the search bar to filter the items displayed in the inventory list. This feature enhances the user experience by allowing them to quickly locate specific items without scrolling through the entire inventory.

Changes Made:
- Implemented search functionality in the inventory module
- Added search bar UI component
- Updated inventory display logic to filter items based on search keywords
- Implemented search query handling and filtering logic
- Updated unit tests to cover the new search functionality

Commit:
Duke_Level10
Commit: Implement graphical user interface for chatbot

Description:
This commit introduces the graphical user interface (GUI) for the chatbot application. The GUI provides users with a visually appealing interface to interact with the chatbot, improving the overall user experience. The chatbot GUI includes input and output text fields, as well as buttons for sending messages and clearing the chat history.

Changes Made:
- Created main GUI window layout using JavaFX
- Implemented text input and output components for chat messages
- Added functionality to send user messages and display bot responses
- Implemented button actions for sending messages and clearing chat history
- Integrated chatbot logic with the GUI components

Commit:
Added : class Assertions

Commit:
Added Assertions.java

Commit:
Added : class Assertions

Commit:
add Assertions.java

Commit:
Merge branch 'A-Assertions' of https://github.com/SamuelZero1/ip into A-Assertions

Commit:
Assertions

Commit:
Minor changes to ChatBotGUI.java

Commit:
Merge branch 'A-CodeQuality'

Commit:
add UserGuide.md

SamuelZero1 committed:
Update README.md

Commit:
modified some classes

Commit:
Merge branch 'master' of https://github.com/SamuelZero1/ip

Commit:
minor changes

SamuelZero1 committed:
Add files via upload

SamuelZero1 committed:
Update README.md

Commit:
update SamuelBot.txt

Commit:
update SamuelBot.txt

Commit:
run SamuelBot.txt

Commit:
Update SamuelBot.txt

Commit:
Update SamuelBot.txt

Commit:
Update SamuelBot.txt

Commit:
Update SamuelBot.txt

Commit:
Update SamuelBot.txt
